### ISSUE TITLE
Sonar 6.7 LTS support

### ISSUE THREAD

**Issue by kingoleg** [AmadeusITGroup/sonar-coverage-evolution#7]
Hi,

Does it work with 6.7 LTS sonar version?

Is there any instructions to run both sonar-stash analysys (-Dsonar.analysis.mode=issues) and this plugin (-Dsonar.analysis.mode=preview?). Or preview is mistake?

Regards,
Oleg

**Comment by t-8ch** [comment]
Hi @kingoleg 
It worked in my tests. Please make sure to specify `-Dsonar.scanAllFiles` for the coverage part.
Both plugins should work absolutely fine together.

**Comment by kingoleg** [comment]
Hi,

I tried it little bit more and found some strange behavior in 1.3 release and in current master version together with 6.7 sonar

**Comment by t-8ch** [comment]
Can you elaborate?

**Comment by kingoleg** [comment]
Sorry, I added a comment related to main plugin here. I created a PR with changes I need. Please, check 

Regarding this plugin. Can you please provide some instruction how to build, setup and run this plugin?

**Comment by t-8ch** [comment]
Building:
`mvn package`

Install:
Copy the jar from `target/` to the extensions folder.

Set up:
Enable the rules as defined by README.md.

Run:
Just run SQ normally.

**Comment by kingoleg** [comment]
It was not easy to find the sonar rule. As coverageEvolution* is just a key, there is not search by key...

**Comment by kingoleg** [comment]
Next, as far as I understand, I should run branch full scan and then PR scan to test this plugin...

**Comment by kingoleg** [comment]
It should not get from server metrics for *.xml file, if coverageEvolution* rule is not active for xml language...

**Comment by kingoleg** [comment]
Ah, no, stop. It totally ignores -Dsonar.branch=blablabla

**Comment by kingoleg** [comment]
https://github.com/AmadeusITGroup/sonar-coverage-evolution/pull/8 fixes branches

**Comment by kingoleg** [comment]
But :) I remember, when this functionality was a part of main plugin, it was a bug with new added file not covered by test. It should blame user!

If we add a file 0 coverage, it reduce total project coverage (if it's not 0, of cause)

**Comment by kingoleg** [comment]
Yes, I see that overall coverage issue is created, but I do not see it in the report or in filtering

```
build	21-Dec-2017 21:15:03	[DEBUG] org.sonarsource.scanner.maven.SonarQubeMojo - 21:15:03.908 Previous/current project-wide coverage: 100.0 / 66.66666666666667
build	21-Dec-2017 21:15:03	[DEBUG] org.sonarsource.scanner.maven.SonarQubeMojo - 21:15:03.909 Creating global coverage issue for org.sonar.api.resources.Project@2f9bd5ac[id=<null>,key=a.a.a,qualifier=TRK]
build	21-Dec-2017 21:15:03	[DEBUG] org.sonarsource.scanner.maven.SonarQubeMojo - 21:15:03.915 RuleKey: coverageEvolution-xml:decreasingOverallLineCoverage
```

**Comment by kingoleg** [comment]
Ah, I understand what is wrong with project overall coverage. For some reason it decided that the main language is xml. And created a rule  coverageEvolution-xml:decreasingOverallLineCoverage. But I expect coverage only for java and typescript. There is no sense for xml coverage :)

**Comment by t-8ch** [comment]
Hi @kingoleg,

thanks for your tests!

I will document the names of the implemented rules.

The plugin should choose the most common language in the repository to create the overall issue for.
(See  https://github.com/AmadeusITGroup/sonar-coverage-evolution/blob/ff59ee80cbe3313df4134788e75d0f775324ec46/src/main/java/org/sonar/plugins/coverageevolution/CoverageRule.java#L66 )

Did you test it with a small example project? In this case, your xml configuration is probably longer than your code.
Unfortunately there are cases where coverage on xml is actually valid :-)
Will the autodetection work for you on larger projects?
Otherwise we would have to add a setting for this, which I don't really like.

**Comment by t-8ch** [comment]
And the language used should probably be logged.

**Comment by kingoleg** [comment]
Hi, @t-8ch 

Yes, I found this `mostCommonLanguage` at the last moment.

As far as I understand, there is not option to add a general rule. not related to language, in sonar. But project-wide evolution rule is not related to one language by nature.

Yes, I tested on small java project with 1 xml (pom.xml) and 2 tiny java classes. Unfornuanly, number of xml lines was bigger then java lines :) 

BTW, do you think it's good idea to calculated this most common language? May be it's better to setup it via settings? In my main project I have 100k+ lines of code. It can take time and this calculated language can be different for each run. It means, I have to add this coverage evolution rule for each language to be sure it works properly.

**Comment by t-8ch** [comment]
Yes, the fact that we can't add an issue without a language and not `generic` language exists prompted the creation of `mostCommonLanguage`.
I don't think perfomance will be a problem (Any perfomance impact will be dwarfed by the cost of talking to SQ for each of the 100k files).

The problem with changing languages may indeed be a problem for certain users.
I will add a setting but keep the fallback to the calculated language, as it should work for most cases.
